data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/stream/StreamWorkerTaskRefresher.java

```diff
@@ -0,0 +1,121 @@
+
+    public StreamWorkerTaskRefresher(final EnhancedSourceCoordinator sourceCoordinator,
+                                     final StreamPartition streamPartition,
+                                     final StreamCheckpointer streamCheckpointer,
+                                     final String s3Prefix,
+                                     final BinlogClientFactory binlogClientFactory,
+                                     final Buffer<Record<Event>> buffer,
+                                     final AcknowledgementSetManager acknowledgementSetManager,
+                                     final ExecutorService executorService,
+                                     final PluginMetrics pluginMetrics) {
+        this.sourceCoordinator = sourceCoordinator;
+        this.streamPartition = streamPartition;
+        this.streamCheckpointer = streamCheckpointer;
+        this.s3Prefix = s3Prefix;
+        this.buffer = buffer;
+        this.pluginMetrics = pluginMetrics;
+        this.acknowledgementSetManager = acknowledgementSetManager;
+        this.binlogClientFactory = binlogClientFactory;
+        this.executorService = executorService;
+        this.credentialsChangeCounter = pluginMetrics.counter(CREDENTIALS_CHANGED);
+        this.taskRefreshErrorsCounter = pluginMetrics.counter(TASK_REFRESH_ERRORS);
+    }
+
+    public static StreamWorkerTaskRefresher create(final EnhancedSourceCoordinator sourceCoordinator,
+                                                   final StreamPartition streamPartition,
+                                                   final StreamCheckpointer streamCheckpointer,
+                                                   final String s3Prefix,
+                                                   final BinlogClientFactory binlogClientFactory,
+                                                   final Buffer<Record<Event>> buffer,
+                                                   final AcknowledgementSetManager acknowledgementSetManager,
+                                                   final ExecutorService executorService,
+                                                   final PluginMetrics pluginMetrics) {
+        return new StreamWorkerTaskRefresher(sourceCoordinator, streamPartition, streamCheckpointer, s3Prefix,
+                binlogClientFactory, buffer, acknowledgementSetManager, executorService, pluginMetrics);
+    }
+
+    public void initialize(RdsSourceConfig sourceConfig) {
+        currentSourceConfig = sourceConfig;
+        refreshTask(sourceConfig);
+    }
+
+    @Override
+    public void update(RdsSourceConfig sourceConfig) {
+        if (basicAuthChanged(sourceConfig.getAuthenticationConfig())) {
+            LOG.info("Database credentials were updated. Refreshing stream worker...");
+            credentialsChangeCounter.increment();
+            try {
+                executorService.shutdownNow();
+                executorService = Executors.newSingleThreadExecutor();
```

Comment:
This can be a lambda function that can be passes to constructor.  ExecutorService can be initialized from this lambda function in Line 63 and here.

Reply:
Thanks! Did the refactor accordingly.